frontend/src/Editor/Viewer.jsx

```diff
@@ -127,16 +127,22 @@ class Viewer extends React.Component {
     });
   };
 
-  fetchOrgEnvironmentVariables = async () => {
+  fetchOrgEnvironmentVariables = async (slug) => {
     const variables = {
       client: {},
       server: {},
     };
-    await orgEnvironmentVariableService.getVariables().then((data) => {
-      data.variables.map((variable) => {
-        variables[variable.variable_type][variable.variable_name] =
-          variable.variable_type === 'server' ? 'HiddenEnvironmentVariable' : variable.value;
-      });
+
+    let variablesResult;
+    if (this.state.currentUser) {
+      variablesResult = await orgEnvironmentVariableService.getVariables();
```

Comment:
It will fail for this condition:
- User logged in to workspace A
- opening public app of workspace B
- `organization-variables` API will return variables of workspace A instead of B

Shall we check if app is public and load the data accordingly instead of checking the availability of `currentUser`?